tools/hlstool/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+set(LLVM_LINK_COMPONENTS
+  Support
+  )
+
+add_llvm_tool(hlstool
+ hlstool.cpp
+)
+llvm_update_compile_flags(hlstool)
+target_link_libraries(hlstool
+  PRIVATE
+
+  CIRCTAffineToPipeline
```

Comment:
We can probably drop some of these libraries to reduce the linking time substantially.

Reply:
Took the lazy route by using
```c++
circt::registerAllPasses();
circt::registerAllDialects(registry);
```
but i agree that link times should be prioritized in this case. Will change.